### PR TITLE
BAEL-4663: Whats new in Java 15

### PR DESCRIPTION
**Pull request by michael-pratt**

### CODE REVIEW

File: core-java-modules/core-java-15/src/main/java/whatsnew/records/Person.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package whatsnew.records;
```
Comment (eric-martin) [inline core-java-modules/core-java-15/src/main/java/whatsnew/records/Person.java:1]: Let's use com.baeldung.whatsnew for the package

Comment (michael-pratt) [reply core-java-modules/core-java-15/src/main/java/whatsnew/records/Person.java:1]: Just pushed an updated branch with updated package names

File: core-java-modules/core-java-15/src/main/java/com/baeldung/whatsnew/sealedclasses/Employee.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.baeldung.whatsnew.sealedclasses;
+
+import java.util.Date;
+
+public non-sealed class Employee extends Person {
+
+    public Date getHiredDate()
+    {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-15/src/main/java/com/baeldung/whatsnew/sealedclasses/Employee.java:8]: Please put the opening brace at the end of the previous line, not on a line by itself.

File: core-java-modules/core-java-15/src/main/java/com/baeldung/whatsnew/sealedclasses/Person.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package com.baeldung.whatsnew.sealedclasses;
+
+import java.util.Date;
+
+public sealed class Person permits Employee, Manager
+{
```
Comment (KevinGilmore) [inline core-java-modules/core-java-15/src/main/java/com/baeldung/whatsnew/sealedclasses/Person.java:6]: move opening brace

File: core-java-modules/core-java-15/src/main/java/com/baeldung/whatsnew/sealedclasses/Person.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    /**
+     * Demonstration of pattern matching for instanceof
+     *
+     * @param person A Person object
+     * @return
+     */
+    public static void patternMatchingDemo(Person person)
+    {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-15/src/main/java/com/baeldung/whatsnew/sealedclasses/Person.java:14]: move opening brace

File: core-java-modules/core-java-15/src/main/java/com/baeldung/whatsnew/sealedclasses/Person.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        if(person instanceof Employee employee)
+        {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-15/src/main/java/com/baeldung/whatsnew/sealedclasses/Person.java:16]: move opening brace

File: core-java-modules/core-java-15/src/main/java/com/baeldung/whatsnew/sealedclasses/Person.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+            Date hiredDate = employee.getHiredDate();
+        }
+
+        if(person instanceof Employee employee && employee.getHiredDate() != null)
+        {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-15/src/main/java/com/baeldung/whatsnew/sealedclasses/Person.java:21]: move opening brace

File: core-java-modules/core-java-15/pom.xml [hunk 7]
```diff
@@ -51,8 +51,8 @@
                 <configuration>
                     <release>${maven.compiler.release}</release>
                     <compilerArgs>--enable-preview</compilerArgs>
-                    <source>15</source>
-                    <target>15</target>
+                    <source>14</source>
+                    <target>14</target>
```
Comment (KevinGilmore) [inline core-java-modules/core-java-15/pom.xml:55]: Is this correct? Why doesn't it use 15?

**Review by KevinGilmore — CHANGES_REQUESTED** [review]
Put opening braces at end of line rather than on their own line. Also check the question about using 14 vs 15 for source and target in the compiler plugin config.

File: core-java-modules/core-java-15/src/main/java/com/baeldung/whatsnew/sealedclasses/Employee.java (hunk 2 above)
Comment (michael-pratt) [reply core-java-modules/core-java-15/src/main/java/com/baeldung/whatsnew/sealedclasses/Employee.java:8]: Done

File: core-java-modules/core-java-15/src/main/java/com/baeldung/whatsnew/sealedclasses/Person.java (hunk 4 above)
Comment (michael-pratt) [reply core-java-modules/core-java-15/src/main/java/com/baeldung/whatsnew/sealedclasses/Person.java:14]: Done

File: core-java-modules/core-java-15/src/main/java/com/baeldung/whatsnew/sealedclasses/Person.java (hunk 6 above)
Comment (michael-pratt) [reply core-java-modules/core-java-15/src/main/java/com/baeldung/whatsnew/sealedclasses/Person.java:21]: Done

File: core-java-modules/core-java-15/pom.xml (hunk 7 above)
Comment (michael-pratt) [reply core-java-modules/core-java-15/pom.xml]: Good question. Assuming a copy/paste error from another module? Regardless I updated to 15.

**Pull request merged**